graphql-schemas/schema.yml

```diff
@@ -1857,6 +1857,45 @@
   - { name: pipelineTemplates, type: PipelinesProviderPipelineTemplates_v1, isRequired: true }
   - { name: deployResources, type: DeployResources_v1 }
 
+- name: VaultInstance_v1
+  isInterface: true
+  interfaceResolve:
+    strategy: fieldMap
+    field: authType
+    fieldMap:
+      approle: VaultInstanceApprole_v1
+      token: VaultInstanceToken_v1
+  fields:
+  - { name: schema, type: string, isRequired: true }
+  - { name: labels, type: json }
+  - { name: name, type: string, isRequired: true }
+  - { name: description, type: string, isRequired: true }
+  - { name: authType, type: string, isRequired: true }
```

Comment:
i'd recommend to create an `auth` section which uses the provider pattern: https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-interface/qontract-reconcile-patterns.md#the-provider-pattern

that way, a vault instance doesn't need to be an interface. it's really only the auth, isn't it?

Reply:
Good point. This approach would be cleaner. I will work on implementing

Reply:
Implemented